packages/js/src/classic-editor/components/seo-analysis.js

```diff
@@ -0,0 +1,203 @@
+ */
+const SynonymsUpsell = () => {
+	const modalProps = {
+		classes: {
+			openButton: "wpseo-keyword-synonyms button-link",
+		},
+		labels: {
+			open: "+ " + __( "Add synonyms", "wordpress-seo" ),
+			modalAriaLabel: sprintf(
+				/* translators: %s expands to 'Yoast SEO Premium'. */
+				__( "Get %s", "wordpress-seo" ),
+				"Yoast SEO Premium"
+			),
+			heading: sprintf(
+				/* translators: %s expands to 'Yoast SEO Premium'. */
+				__( "Get %s", "wordpress-seo" ),
+				"Yoast SEO Premium"
+			),
+		},
+	};
+
+	// Defaults to metabox.
+	const link    = wpseoAdminL10n[ "shortlinks.upsell.metabox.focus_keyword_synonyms_link" ];
+	const buyLink = wpseoAdminL10n[ "shortlinks.upsell.metabox.focus_keyword_synonyms_button" ];
+
+	return (
+		<Modal { ...modalProps }>
+			<ModalContainer>
+				<ModalIcon icon={ YoastSeoIcon } />
+				<h2>{ __( "Would you like to add keyphrase synonyms?", "wordpress-seo" ) }</h2>
+
+				<KeywordSynonyms link={ link } buyLink={ buyLink } />
+			</ModalContainer>
+		</Modal>
+	);
+};
+
+/**
+ * The multiple keywords upsell modal.
+ *
+ * @returns {wp.Element} A modalButtonContainer component with the modal for a multiple keywords upsell.
+ */
+const MultipleKeywordsUpsell = () => {
+	const modalProps = {
+		classes: {
+			openButton: "wpseo-multiple-keywords button-link",
+		},
+		labels: {
+			open: "+ " + __( "Add related keyphrase", "wordpress-seo" ),
+			modalAriaLabel: sprintf(
+				/* translators: %s expands to 'Yoast SEO Premium'. */
+				__( "Get %s", "wordpress-seo" ),
+				"Yoast SEO Premium"
+			),
+			heading: sprintf(
+				/* translators: %s expands to 'Yoast SEO Premium'. */
+				__( "Get %s", "wordpress-seo" ),
+				"Yoast SEO Premium"
+			),
+		},
+	};
+
+	// Defaults to metabox
+	const link    = wpseoAdminL10n[ "shortlinks.upsell.metabox.focus_keyword_additional_link" ];
+	const buyLink = wpseoAdminL10n[ "shortlinks.upsell.metabox.focus_keyword_additional_button" ];
+
+	return (
+		<Modal { ...modalProps }>
+			<ModalContainer>
+				<ModalIcon icon={ YoastSeoIcon } />
+				<h2>{ __( "Would you like to add a related keyphrase?", "wordpress-seo" ) }</h2>
+				<MultipleKeywords
+					link={ link }
+					buyLink={ buyLink }
+				/>
+			</ModalContainer>
+		</Modal>
+	);
+};
+
+/**
+ * The SEO Analysis component.
+ *
+ * @param {object} props The component's props.
+ *
+ * @returns {wp.Element} The SEO Analysis component.
+ */
+const SeoAnalysisContainer = props => {
+	const { shouldUpsell, shouldUpsellWordFormRecognition } = props;
+
+	const focusKeyphrase = useSelect( select => select( SEO_STORE_NAME ).selectKeyphrase() );
+	const seoScore = useSelect( select => select( SEO_STORE_NAME ).selectSeoScore() );
+
+	const score = getIndicatorForScore( seoScore );
+
+	if ( score.className !== "loading" && focusKeyphrase === "" ) {
+		score.className = "na";
+		score.screenReaderReadabilityText = __( "Enter a focus keyphrase to calculate the SEO score", "wordpress-seo" );
+	}
+
+	return (
+		<>
```

Comment:
We might want to use the full `Fragment` instead of the shorthand. Something about our release tooling not handling that correctly. Enrico ran into that for the 17.7 release.
I'm not sure about this, but can't hurt to use the full for safety, right?